### ISSUE TITLE
[Bug]: Reason parameter in BanAsync() method is ignored

### ISSUE THREAD

**Issue by Liege72** [RogueException/Discord.Net#2626]
### Check The Docs

- [X] I double checked the docs and couldn't find any useful information.

### Verify Issue Source

- [X] I verified the issue was caused by Discord.Net.

### Check your intents

- [X] I double checked that I have the required intents.

### Description

When banning a user with the `SocketGuildUser.BanAsync(pruneDays, reason)` method, the reason parameter is ignored. Ignored being: the reason is not shown in audit logs or in the ban details in the Ban tab of your guild.

A simple fix to this issue would be to use `RequestOptions()` and `AuditLogReason`, but @quinchs told me to report this :)

### Version

3.8.1

### Logs

There are no logs.

### Sample

```cs
var user = Context.Guild.GetUser(userid);
await user.BanAsync(0, "reason");
```

### Packages

Microsoft.ML 2.0.0
MongoDB.Bson 2.18.0
MongoDB.Driver 2.18.0
StackExchange.Redis 2.6.70

**Comment by cjbonn** [comment]
This was already fixed in #2540

**Comment by quinchs** [comment]
fixing pr was merged and included in 3.10.0
